09_json/willshen8/pkg/puppy/json/jsonifyPuppy.go

```
@@ -0,0 +1,22 @@
+package puppy
+
+import (
+	"encoding/json"
+
+	puppy "github.com/anz-bank/go-course/09_json/willshen8/pkg/puppy"
+	store "github.com/anz-bank/go-course/09_json/willshen8/pkg/puppy/store"
+)
+
+func UnmarshalPuppy(jsonData []byte, decodedPuppies *[]puppy.Puppy) {
+	unMarshalError := json.Unmarshal(jsonData, &decodedPuppies)
+	if unMarshalError != nil {
+		panic(unMarshalError)
```

IMO panic should be left to main(), I'd return an error here (and below).

---

Thanks, I agree. Then you can use the return value to see if it's successful.